Suggest to move/copy use statement from parent into inline module declaration

Given the below Rust code, we can probably produce a more user friendly error message than today's "failed to resolve. Use of undeclared type or module `File`".

```rust
use std::fs::File;

mod foo {
    fn some_function(f: File) {}
}
```

It'd be nice to suggest something like "Possibly move `use std::fs::File` from outer scope into module `foo`". This will help beginners understand how to deal with the `File` use in the parent module not being in scope within the child module.

@Mark-Simulacrum I think like you're using an old `rustc` -- that hasn't been the error since `1.14` or earlier. The error on beta and nightly is:
```
error[E0412]: cannot find type `File` in this scope
 --> example.rs:4:25
  |
4 |     fn some_function(f: File) {}
  |                         ^^^^ not found in this scope
  |
  = help: possible candidate is found in another module, you can import it into scope:
            `use std::fs::File;`
```

If the user adds `use std::fs::File;` as suggested, the the top-level import will be an unused import warning.

cc @petrochenkov

--- a/src/librustc_resolve/diagnostics.rs
+++ b/src/librustc_resolve/diagnostics.rs
@@ -838,6 +838,32 @@ trait Foo {
 
 fn foo<T>(x: T) {} // ok!
 ```
+
+Another case that causes this error is when a type is imported into a parent
+module. To fix this, you can follow the suggestion and use File directly or
+`use super::File;` which will import the types from the parent namespace. An
+example that causes this error is below:
+
+```compile_fail,E0412
+use std::fs::File;
+
+mod foo {
+    fn some_function(f: File) {}
+}
+```
+
+```
+use std::fs::File;
+
+mod foo {
+    // either
+    use super::File;
+    // or
+    // use std::fs::File;
+    fn foo(f: File) {}
+}
+# fn main() {} // don't insert it for us; that'll break imports
+```
 "##,
 
 E0415: r##"

